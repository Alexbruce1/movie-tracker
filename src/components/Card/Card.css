.Card {
	align-items: center;
	background-color: #fcfcfc;
	background-position: top center;
	background-size: cover;
	box-shadow: 0 8px 1rem rgba(0, 0, 0, .5);
	cursor: pointer;
	display: flex;
	flex-direction: column;
	height: 20rem;
	justify-content: space-between;
	margin: 1rem auto;
	padding: 0 1rem;
	width: 93%;
	transition-duration: .1s;
}

.Card:hover {
	box-shadow: 0 12px 1rem rgba(0, 0, 0, .5);
}

.card-inner-wrapper {
	width: calc(100% + 2rem);
}

.Card p {
	display: none;
}

.Card:hover .movie-title {
	background: linear-gradient(to bottom, rgba(0, 0, 0, .9), rgba(0, 0, 0, .6), transparent);
}

.movie-rating {
	font-size: 1rem;
	height: 2.5rem;
	margin-left: 1rem;
	width: 2.5rem;
}

.card-favorite-button {
	background-color: rgba(0, 0, 0, .3);
	/* border: 2px solid #fcfcfc; */
	border-radius: 50%;
	/* box-shadow: inset 0 0 6px rgba(0, 0, 0, .6); */
	cursor: pointer;
	height: 2.5rem;
	margin-right: 1rem;
	outline: none;
	width: 2.5rem;
}

.card-favorite-button img {
	filter: invert(90%);
}

.movie-title {
	background: linear-gradient(to bottom, rgba(0, 0, 0, .8), rgba(0, 0, 0, .5), transparent);
	color: #fcfcfc;
	display: inline-flex;
	font-size: 2rem;
	justify-content: space-between;
	height: 4rem;
	padding: .5rem 0;
	margin: 0;
	overflow: hidden;
	width: calc(100% + 1px);
}

.Card p {
	margin: 0;
}

.poster-image {
	width: 15rem;
}

@media screen and (max-width: 1600px) {
  .Card {
		height: 17rem;
		width: 25rem;
	}

	.movie-title {
		font-size: 1.5rem;
		width: 100%;
	}
}

@media screen and (max-width: 900px) {
	.Card {
		height: 13rem;
		width: 20rem;
	}
}

@media screen and (max-width: 730px) {
	.Card {
		border-radius: 0;
		height: 15rem;
		margin: 0;
		width: 100%;
	}

	.movie-title {
		border-radius: 0;
	}
}