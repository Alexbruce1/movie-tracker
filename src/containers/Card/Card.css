.Card {
	align-items: center;
	background-color: #fcfcfc;
	background-position: top center;
	background-size: cover;
	box-shadow: 0 8px 1rem rgba(0, 0, 0, .5);
	cursor: pointer;
	display: flex;
	flex-direction: column;
	height: 20rem;
	justify-content: space-between;
	margin: 1rem auto;
	padding: 0;
	width: 100%;
	transition-duration: .1s;
}

.Card:hover {
	box-shadow: 0 12px 1rem rgba(0, 0, 0, .5);
}

.card-inner-wrapper {
	width: 100%;
}

.Card p {
	display: none;
}

.Card:hover .movie-title {
	background: linear-gradient(to bottom, rgba(0, 0, 0, .9), rgba(0, 0, 0, .6), transparent);
}

.movie-rating {
	font-size: 1rem;
	height: 2.5rem;
	margin-left: 1rem;
	width: 2.5rem;
}

.card-favorite-button {
	background-color: #222;
	border: none;
  bottom: 8px;
  box-shadow: inset 0 0 4px rgba(0, 0, 0, .4);
  clip-path: polygon(0 0, 100% 100%, 100% 0);
	cursor: pointer;
  height: 70px;
	outline: none;
	position: relative;
	right: 1px;
	transition-duration: .2s;
  width: 70px;
}

.card-favorite-button:hover {
	background-color: #444;
}

.card-favorite-button img {
	bottom: 15px;
	height: 2rem;
	left: 15px;
	position: relative;
}

.fav-btn-active {
	background-color: #444;
}

.fav-btn-active:hover {
	background-color: #222;
}

.movie-title {
	background: linear-gradient(to bottom, rgba(0, 0, 0, .8), rgba(0, 0, 0, .5), transparent);
	color: #fcfcfc;
	display: inline-flex;
	font-size: 2rem;
	justify-content: space-between;
	height: 4rem;
	padding: .5rem 0;
	text-shadow: 0 0 2rem rgba(0, 0, 0, .9);
	margin: 0;
	overflow: hidden;
	width: calc(100% + 1px);
}

.Card p {
	margin: 0;
}

.poster-image {
	display: none;
	width: 15rem;
}

/* --------------- EXPANDED CARD --------------- */

.expanded-background {
	display: none;
}

.expanded-card {
	height: 80%;
	margin-left: 10vw;
	position: fixed;
	width: 80%;
	z-index: 3;
}

.expanded-card .card-inner-wrapper {
	/* box-shadow: inset 0 0 5rem rgba(255, 255, 255, .3); */
}

.expanded-card .expanded-background {
	background-color: rgba(0, 0, 0, .4);
	/* background: radial-gradient(transparent 50%, rgba(0, 0, 0, .8)); */
	display: initial;
	height: 100vh;
	left: 0;
	position: fixed;
	top: 0;
	width: 100vw;
	z-index: -1;
}

.expanded-card .movie-title {
	padding-top: 2rem;
}

.expanded-card .card-favorite-button {
	background-color: transparent;
	width: 10%;
}

.expanded-card .card-inner-wrapper {
	background-color: transparent;
	height: 100%;
}

.expanded-lower-container {
	/* background-color: chartreuse; */
	display: inline-flex;
	justify-content: space-around;
	height: 80%;
}

.expanded-card .poster-image {
	box-shadow: 0 0 3rem #666;
	border-radius: 5px;
	display: inline;
	height: 70%;
	margin: 4rem 3rem;
	width: auto;
}

.expanded-card p {
	/* background-color: magenta; */
	color: white;
	display: inline;
	padding-top: 4rem;
}

.card-overview {
	font-size: 2rem;
	text-align: justify;
	width: 45%;
}

.card-release-date {
	font-size: 1.5rem;
	text-align: end;
}

/* --------------- MEDIA QUERIES --------------- */

@media screen and (max-width: 1600px) {
  .Card {
		height: 17rem;
	}

	.card-favorite-button {
		right: 0;
	}

	.movie-title {
		font-size: 1.5rem;
		width: 100%;
	}

	.expanded-card {
		height: 70%;
		margin-top: 2rem;
	}

	.expanded-card p {
		font-size: 1.5rem;
	}

	.expanded-card .poster-image {
		height: unset;
		margin: 4rem 0;
		width: 10rem;
	}
}

@media screen and (max-width: 900px) {
	.Card {
		height: 13rem;
	}

	.card-favorite-button {
		left: 1px;
	}

	.expanded-card p {
		font-size: 15px;
	}
}

@media screen and (max-width: 730px) {
	.Card {
		border-radius: 0;
		height: 15rem;
		margin: 0;
		width: 100%;
	}

	.movie-title {
		border-radius: 0;
	}
}





.rating-container {
  background-color: #fcfcfc;
  border-radius: 2rem;
  height: 2rem;
  margin: 3rem auto;
  padding: 2px;
  width: 100%;
}

.rating-bar {
  /* background-color: #51B04D; */
	border-radius: 1rem;
	font-size: 1rem;
	font-weight: 700;
	line-height: 200%;
	text-align: center;
  height: 32px;
}