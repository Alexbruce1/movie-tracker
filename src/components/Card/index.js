import React from 'react';
import './Card.css';
import star from '../../images/star-clear.svg';
import filledStar from '../../images/star.svg';
import { toggleFavorite, addFavorite, removeFavorite } from '../../actions/userActions';
import { connect } from 'react-redux';
import * as API from '../../utilities/API'

const Card = ({ movie, user, favorites, changeFavorite, addToFavorites, removeFromFavorites }) => {
	const handleFavorite = (movie) => {
		const { id, isFavorite } = movie
	 	changeFavorite(id)
	 	if(isFavorite) {
			removeFromFavorites(movie)
	 		API.removeFavorite(movie, user)
	 	} else {
	 		addToFavorites(movie)
			API.addFavorite(movie, user)
	 	}
	}

	return(
		<div className='Card' style={{backgroundImage: 'url(' + movie.backdrop + ')'}}>
			<div className="card-inner-wrapper">
				<h3 className="movie-title" >
					<div className="movie-rating" >Rating {movie.rating}/10</div>
					{movie.title}
					<button 
						className={`card-favorite-button
							${movie.isFavorite || favorites.includes(movie)
								? 'fav-btn-active'
								: 'fav-btn-inactive'}`} 
						onClick={() => handleFavorite(movie)} 
					>
						<img alt="" src={star} />
					</button>
				</h3>
				{/* <img  */}
					{/* src={movie.poster}  */}
					{/* className='poster-image'/> */}
				<p>{movie.overview}</p>
				<p>Opens: {movie.releaseDate}</p>
				<p>Viewer Rating: {movie.rating}</p>
			</div>
	  </div>
  )
}

export const mapStateToProps = (state) => ({
	user: state.user,
	favorites: state.favorites
})

export const mapDispatchToProps = (dispatch) => ({
	changeFavorite: (id) => dispatch(toggleFavorite(id)),
	addToFavorites: (movie) => dispatch(addFavorite(movie)),
	removeFromFavorites: (movie) => dispatch(removeFavorite(movie))
})

export default connect(mapStateToProps, mapDispatchToProps)(Card);
