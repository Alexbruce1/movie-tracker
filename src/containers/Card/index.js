import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { toggleFavorite } from '../../actions/movieActions';
import { removeFavorite, addFavorite } from '../../utilities/API';
import star from '../../images/star-clear.svg';
import './Card.css';

export class Card extends Component {
	constructor() {
		super();
		this.state = {
			expanded: false
		}
	}

	handleFavorite = async (movie) => {
    const { user, toggleFavorite } = this.props;
		const { id, favorite } = movie;
    if (!user.loggedIn) {
      return undefined;
    }
    if (favorite) {
      removeFavorite(movie, user);
    } else {
      addFavorite(movie, user);
    }
      toggleFavorite(id);
    }

  handleExpand = (e) => {
    if (!e.target.className.includes('card-favorite-button')) {
      this.setState({
        expanded: !this.state.expanded
      })
    }
  }

  render() {
    const { movie, user } = this.props;
    const { expanded } = this.state;

    return(
			<div
				className={expanded ? 'Card' : 'Card card-expanded'}
				style={{backgroundImage: 'url(' + movie.backdrop + ')'}}
        onClick={this.handleExpand}>
        <div className="card-inner-wrapper">
          <h3 className="movie-title" >
            <div className="movie-rating" >Rating {movie.rating}/10</div>
            {movie.title}
            <button
              className={`card-favorite-button
                ${movie.favorite && user.loggedIn
                  ? 'fav-btn-active'
                  : 'fav-btn-inactive'}`}
              onClick={() => this.handleFavorite(movie)}>
              <img className="card-favorite-button-image" alt="" src={movie.favorite ? filledStar : star} />
            </button>
          </h3>
          <div className="expanded-lower-container">
            <img 
              src={movie.poster} 
              alt=""
              className='poster-image'/>
            <p className="card-overview">
              <strong>Description</strong>
              <br/>
              {movie.overview}
            </p>
            <p className="card-release-date">
              <strong>Release Date</strong>
              <br/>
              {movie.releaseDate}</p>
          </div>
        </div>
      </div>
    )
  }
}

export const mapStateToProps = ({ user }) => ({ user });

export const mapDispatchToProps = (dispatch) => ({
	toggleFavorite: (movieId) => dispatch(toggleFavorite(movieId)),
});

Card.propTypes = {
  movie: PropTypes.object.isRequired,
}

export default connect(mapStateToProps, mapDispatchToProps)(Card);