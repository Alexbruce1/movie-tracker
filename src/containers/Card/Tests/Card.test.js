import React from 'react';
import { shallow } from 'enzyme';
import Card from '../';

describe('Card', () => {
  let wrapper;
  beforeEach(() => {
    const mockMovie = {
      movie: {
        title: 'Movie Title',
        poster: 'Movie Poster',
        overview: 'Movie Overview',
        releaseData: 'Movie Release Date',
        rating: 'Movie Rating'
      }}
    wrapper = shallow(<Card movie={mockMovie}/>);
  });

  it('Should render like snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
