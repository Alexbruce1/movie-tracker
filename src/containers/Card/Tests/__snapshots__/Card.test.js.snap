// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    movie={
      Object {
        "movie": Object {
          "favorite": false,
          "id": 123456,
          "overview": "Movie Overview",
          "poster": "Movie Poster",
          "rating": "Movie Rating",
          "releaseData": "Movie Release Date",
          "title": "Movie Title",
        },
      }
    } 
    toggleFavorite={[MockFunction]}
    user={
      Object {
        "id": 31,
        "loggedIn": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="expanded-background"
        />,
        <div
          className="card-inner-wrapper"
        >
          <h3
            className="movie-title"
          >
            <div
              className="movie-rating"
            >
              <div>
                Rating 
              </div>
            </div>
            <button
              className="card-favorite-button
                fav-btn-inactive"
              onClick={[Function]}
            >
              <img
                alt=""
                className="card-favorite-button-star"
                src="star-clear.svg"
              />
            </button>
          </h3>
          <div
            className="expanded-lower-container"
          >
            <img
              alt=""
              className="poster-image"
            />
            <div
              className="lower-middle-expanded"
            >
              <p
                className="card-overview"
              >
                <strong>
                  Description
                </strong>
                <br />
              </p>
            </div>
            <p
              className="card-release-date"
            >
              <strong>
                Release Date
              </strong>
              <br />
            </p>
          </div>
        </div>,
      ],
      "className": "Card",
      "onClick": [Function],
      "style": Object {
        "backgroundImage": "url(undefined)",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "expanded-background",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="movie-title"
            >
              <div
                className="movie-rating"
              >
                <div>
                  Rating 
                </div>
              </div>
              <button
                className="card-favorite-button
                fav-btn-inactive"
                onClick={[Function]}
              >
                <img
                  alt=""
                  className="card-favorite-button-star"
                  src="star-clear.svg"
                />
              </button>
            </h3>,
            <div
              className="expanded-lower-container"
            >
              <img
                alt=""
                className="poster-image"
              />
              <div
                className="lower-middle-expanded"
              >
                <p
                  className="card-overview"
                >
                  <strong>
                    Description
                  </strong>
                  <br />
                </p>
              </div>
              <p
                className="card-release-date"
              >
                <strong>
                  Release Date
                </strong>
                <br />
              </p>
            </div>,
          ],
          "className": "card-inner-wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="movie-rating"
                >
                  <div>
                    Rating 
                  </div>
                </div>,
                undefined,
                <button
                  className="card-favorite-button
                fav-btn-inactive"
                  onClick={[Function]}
                >
                  <img
                    alt=""
                    className="card-favorite-button-star"
                    src="star-clear.svg"
                  />
                </button>,
              ],
              "className": "movie-title",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div>
                    Rating 
                  </div>,
                  "className": "movie-rating",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Rating ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Rating ",
                    undefined,
                  ],
                  "type": "div",
                },
                "type": "div",
              },
              undefined,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt=""
                    className="card-favorite-button-star"
                    src="star-clear.svg"
                  />,
                  "className": "card-favorite-button
                fav-btn-inactive",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "card-favorite-button-star",
                    "src": "star-clear.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "button",
              },
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt=""
                  className="poster-image"
                />,
                <div
                  className="lower-middle-expanded"
                >
                  <p
                    className="card-overview"
                  >
                    <strong>
                      Description
                    </strong>
                    <br />
                  </p>
                </div>,
                <p
                  className="card-release-date"
                >
                  <strong>
                    Release Date
                  </strong>
                  <br />
                </p>,
              ],
              "className": "expanded-lower-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "poster-image",
                  "src": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="card-overview"
                    >
                      <strong>
                        Description
                      </strong>
                      <br />
                    </p>,
                    false,
                  ],
                  "className": "lower-middle-expanded",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <strong>
                          Description
                        </strong>,
                        <br />,
                        undefined,
                      ],
                      "className": "card-overview",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Description",
                        },
                        "ref": null,
                        "rendered": "Description",
                        "type": "strong",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": "br",
                      },
                      undefined,
                    ],
                    "type": "p",
                  },
                  false,
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <strong>
                      Release Date
                    </strong>,
                    <br />,
                    undefined,
                  ],
                  "className": "card-release-date",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Release Date",
                    },
                    "ref": null,
                    "rendered": "Release Date",
                    "type": "strong",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  undefined,
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="expanded-background"
          />,
          <div
            className="card-inner-wrapper"
          >
            <h3
              className="movie-title"
            >
              <div
                className="movie-rating"
              >
                <div>
                  Rating 
                </div>
              </div>
              <button
                className="card-favorite-button
                fav-btn-inactive"
                onClick={[Function]}
              >
                <img
                  alt=""
                  className="card-favorite-button-star"
                  src="star-clear.svg"
                />
              </button>
            </h3>
            <div
              className="expanded-lower-container"
            >
              <img
                alt=""
                className="poster-image"
              />
              <div
                className="lower-middle-expanded"
              >
                <p
                  className="card-overview"
                >
                  <strong>
                    Description
                  </strong>
                  <br />
                </p>
              </div>
              <p
                className="card-release-date"
              >
                <strong>
                  Release Date
                </strong>
                <br />
              </p>
            </div>
          </div>,
        ],
        "className": "Card",
        "onClick": [Function],
        "style": Object {
          "backgroundImage": "url(undefined)",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "expanded-background",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="movie-title"
              >
                <div
                  className="movie-rating"
                >
                  <div>
                    Rating 
                  </div>
                </div>
                <button
                  className="card-favorite-button
                fav-btn-inactive"
                  onClick={[Function]}
                >
                  <img
                    alt=""
                    className="card-favorite-button-star"
                    src="star-clear.svg"
                  />
                </button>
              </h3>,
              <div
                className="expanded-lower-container"
              >
                <img
                  alt=""
                  className="poster-image"
                />
                <div
                  className="lower-middle-expanded"
                >
                  <p
                    className="card-overview"
                  >
                    <strong>
                      Description
                    </strong>
                    <br />
                  </p>
                </div>
                <p
                  className="card-release-date"
                >
                  <strong>
                    Release Date
                  </strong>
                  <br />
                </p>
              </div>,
            ],
            "className": "card-inner-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="movie-rating"
                  >
                    <div>
                      Rating 
                    </div>
                  </div>,
                  undefined,
                  <button
                    className="card-favorite-button
                fav-btn-inactive"
                    onClick={[Function]}
                  >
                    <img
                      alt=""
                      className="card-favorite-button-star"
                      src="star-clear.svg"
                    />
                  </button>,
                ],
                "className": "movie-title",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <div>
                      Rating 
                    </div>,
                    "className": "movie-rating",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Rating ",
                        undefined,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Rating ",
                      undefined,
                    ],
                    "type": "div",
                  },
                  "type": "div",
                },
                undefined,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt=""
                      className="card-favorite-button-star"
                      src="star-clear.svg"
                    />,
                    "className": "card-favorite-button
                fav-btn-inactive",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "",
                      "className": "card-favorite-button-star",
                      "src": "star-clear.svg",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "button",
                },
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt=""
                    className="poster-image"
                  />,
                  <div
                    className="lower-middle-expanded"
                  >
                    <p
                      className="card-overview"
                    >
                      <strong>
                        Description
                      </strong>
                      <br />
                    </p>
                  </div>,
                  <p
                    className="card-release-date"
                  >
                    <strong>
                      Release Date
                    </strong>
                    <br />
                  </p>,
                ],
                "className": "expanded-lower-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "poster-image",
                    "src": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <p
                        className="card-overview"
                      >
                        <strong>
                          Description
                        </strong>
                        <br />
                      </p>,
                      false,
                    ],
                    "className": "lower-middle-expanded",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <strong>
                            Description
                          </strong>,
                          <br />,
                          undefined,
                        ],
                        "className": "card-overview",
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "Description",
                          },
                          "ref": null,
                          "rendered": "Description",
                          "type": "strong",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {},
                          "ref": null,
                          "rendered": null,
                          "type": "br",
                        },
                        undefined,
                      ],
                      "type": "p",
                    },
                    false,
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <strong>
                        Release Date
                      </strong>,
                      <br />,
                      undefined,
                    ],
                    "className": "card-release-date",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Release Date",
                      },
                      "ref": null,
                      "rendered": "Release Date",
                      "type": "strong",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "br",
                    },
                    undefined,
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
